lang/pkl/Portfile

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               java 1.0
+PortGroup               github 1.0
+
+github.setup            apple pkl 0.25.1
+
+java.version            17+
+java.fallback           openjdk17
+
+categories              lang
+maintainers             {sub-pop.net:link @subpop} openmaintainer
+
+license                 Apache-2.0
+
+description             A configuration as code language with rich validation and tooling
+long_description        Pkl (pronounced Pickle) is an embeddable configuration \
+                        language which provides rich support for data templating \
+                        and validation. It can be used from the command line, \
+                        integrated in a build pipeline, or embedded in a program. \
+                        Pkl scales from small to large, simple to complex, ad-hoc \
+                        to repetitive configuration tasks.
+
+checksums               rmd160  1e822e7a57050b392a216d9a808283d42df3801b \
+                        sha256  f1588550d820e226b9c387abcffae50c4c64ee313a872acdefbe2f61b6f60e84 \
+                        size    2769668
+
+if {${os.arch} eq "arm"} {
+patchfiles              graalVm23.patch
```

Comment:
where does this patch come from? There is no link or any explanation and ideally we don't want to carry forward such large patches... if upstream thinks it is important they probably should release a new version.

Reply:
also this should be included in the identical `if-statement` below, no need to duplicate it

Reply:
This patch actually comes from within the source tree. While investigating how
to build this program, I discovered that their CI build system applies this
patch to update the version of GraalVM if the architecture is ARM. I'm pretty
sure it fixes a build error on the ARM variant, but now that I have a working
Portfile, I'll try removing the patch and see if I can make it build without it
on ARM.

Reply:
I took a different approach in the latest revision of this commit. Rather than treating the graalVm23.patch file as a patch carried by MacPorts, I apply the patch from *within* the source tree conditionally when building on arm in a `post-patch` phase. This feels more appropriate anyway, since that patch file is actually part of the build process and not some changes that MacPorts is responsible for.

Reply:
> I took a different approach in the latest revision of this commit. Rather than treating the graalVm23.patch file as a patch carried by MacPorts, I apply the patch from _within_ the source tree conditionally when building on arm in a `post-patch` phase. This feels more appropriate anyway, since that patch file is actually part of the build process and not some changes that MacPorts is responsible for.

Sounds good to me

Reply:
> I took a different approach in the latest revision of this commit. Rather than treating the graalVm23.patch file as a patch carried by MacPorts, I apply the patch from *within* the source tree conditionally when building on arm in a `post-patch` phase. This feels more appropriate anyway, since that patch file is actually part of the build process and not some changes that MacPorts is responsible for.

Looks fine to me